presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkBroadcastTableCacheManager.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution;
+
+import com.facebook.presto.common.Page;
+import com.facebook.presto.execution.StageId;
+import com.facebook.presto.spi.plan.PlanNodeId;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkBroadcastTableCacheManager
+{
+    // Currently we cache HT from a single stage. When a task from another stage is scheduled, the cache will be cleared
+    private final Map<BroadcastTableCacheKey, List<List<Page>>> cache = new HashMap<>();
+    private long cacheSizeInBytes;
+
+    public void removeCachedTablesForStagesOtherThan(StageId stageId)
```

Comment:
as an afterthought: Although the cache is currently used always from a single thread, since it is a singleton it might be accidentally used from different threads. I would recommend you protecting all the public methods with `synchronized` to be on the safe side.